### DISCUSSION TITLE
what format is the config file?

### DISCUSSION THREAD

**Discussion by OneCDOnly** [pyload/pyload#4270] [Help] [answered]
Hi all. :)

I'm hoping someone can please advise of the configuration file format for pyLoad. It's not a format I've seen before. 

Example:
```
version: 2

download - "Download":
	int chunks : "Maximum connections for one download" = 3
	time end_time : "End time" = 0:00
	ip interface : "Download interface to bind (IP Address)" =
	bool ipv6 : "Allow IPv6" = False
	bool limit_speed : "Limit download speed" = False
	int max_downloads : "Maximum parallel downloads" = 3
	int max_speed : "Maximum download speed in KiB/s" = -1
	bool skip_existing : "Skip already existing files" = False
	time start_time : "Start time" = 0:00

general - "General":
	debug;trace;stack debug_level : "Debug level" = trace
	bool debug_mode : "Debug mode" = False
```
I need to extract specific values from within BASH, so I'm wondering if a parser already exists to read it. 

Thank you.

**Comment by GammaC0de** [comment, marked as answer]
First line is file format version (always 2).
The has file sections, each section line can be parsed using regex (python regex notation) `(.+?)\s+-\s+"(.+?)":`,
first group in the regex is the section name, group 2 is the section title.

Each section contains configuration line `\s*(?P<T>.+?)\s+(?P<N>[^ ]+?)\s*:\s*"(?P<D>.+?)"\s*=\s?(?P<V>.*)`
Group T - type (bool, int, ip, time , list of values)
Group N - name
Group D - description
Group V - value

**Reply by OneCDOnly** [reply]
Thanks @GammaC0de , that helps a lot. Appreciate the regexes. 👍🏽 

It's a really neat format. I like that it includes the variable type. Did you create it?

**Reply by GammaC0de** [reply]
> I like that it includes the variable type. Did you create it?

No, this was developed many years ago by former pyload maintainers, I fixed some bugs in the parser and created the regexes .

**Reply by OneCDOnly** [reply]
Cheers mate! :)
